trin-core/src/utp/stream.rs

```diff
@@ -174,60 +152,43 @@ impl UtpListener {
         }
     }
 
-    /// Process uTP TalkReq
+    /// Process uTP TalkReq packets
     async fn process_utp_request(&mut self, request: TalkRequest) {
         let payload = request.body();
         let node_id = request.node_id();
 
         match Packet::try_from(payload) {
             Ok(packet) => {
-                self.packet_tx.send(packet.clone()).unwrap();
-
-                let connection_id = packet.connection_id();
-
                 match packet.get_type() {
                     PacketType::Reset => {
-                        let key_fn =
-                            |offset| ConnectionKey::new(*node_id, connection_id - 1 + offset);
-                        let f = |conn: &&mut UtpSocket| -> bool {
-                            conn.sender_connection_id == connection_id
-                        };
-
-                        if let Some(conn) = self.utp_connections.get_mut(&key_fn(1)) {
-                            conn.state = SocketState::Closed;
-                        } else if let Some(conn) =
-                            self.utp_connections.get_mut(&key_fn(2)).filter(f)
-                        {
-                            conn.state = SocketState::Closed;
-                        } else if let Some(conn) =
-                            self.utp_connections.get_mut(&key_fn(0)).filter(f)
-                        {
-                            conn.state = SocketState::Closed;
+                        if let Some(conn) = self.utp_connections.get_mut(node_id) {
+                            if conn.discv5_tx.send(packet).is_ok() {
+                                let mut buf = [0; BUF_SIZE];
+                                if let Err(msg) = conn.recv(&mut buf).await {
+                                    error!("Unable to receive uTP RESET packet: {msg}")
+                                }
+                            } else {
+                                error!("Unable to send RESET packet to uTP stream handler")
+                            }
                         }
```

Comment:
Do we forget to set the socket state to `SocketState::Closed`?

Reply:
No, the code here was leftover from some previous implementation and not up to date.
Now, we change the socket state implicitly when we receive and handle that packet (`conn.recv()` and in `handle_packet()`).